internal/cmd/db.go

```diff
@@ -127,8 +130,64 @@ func getHost() string {
 	return host
 }
 
+type latMap struct {
+	id  string
+	lat *time.Duration
+}
+
 func locations(client *turso.Client) (map[string]string, error) {
 	settings, _ := settings.ReadSettings()
+	return readLocations(settings, client)
+}
+
+func latencies(client *turso.Client) (map[string]int, error) {
+	settings, _ := settings.ReadSettings()
+	locations, err := readLocations(settings, client)
+	if err != nil {
+		return nil, err
+	}
+
+	if latencies := settings.LatencyCache(); latencies != nil {
+		return latencies, nil
+	}
+
+	var wg sync.WaitGroup
+	latencies := make(map[string]int)
+	c := make(chan latMap)
+	for id := range locations {
+		for i := 1; i < 3; i++ {
+			wg.Add(1)
+			go func(id string) {
+				defer wg.Done()
+				dur := probeLocation(id)
+				c <- latMap{id: id, lat: dur}
+
+			}(id)
+		}
+	}
+
+	go func() {
+		for kvp := range c {
+			if kvp.lat != nil {
+				measure := int(kvp.lat.Milliseconds())
+				lat, ok := latencies[kvp.id]
+				if ok {
+					latencies[kvp.id] = int(math.Min(float64(lat), float64(measure)))
+				} else {
+					latencies[kvp.id] = measure
+				}
+			}
+		}
+	}()
```

Comment:
Please do this synchronously or synchronize it through the wait group.

Reply:
why are you worried? It's a single writer.

But will change.

Reply:
Because there is a race condition between you writing the results to the map and writing the map to the cache.
I may have misunderstood the code though.